security/pfSense-pkg-tinc/files/usr/local/pkg/tinc.inc

```diff
@@ -0,0 +1,275 @@
+<?php
+/*
+	tinc.inc
+	part of pfSense (https://www.pfSense.org/)
+	Copyright (C) 2012-2015 ESF, LLC
+	All rights reserved.
+
+	Redistribution and use in source and binary forms, with or without
+	modification, are permitted provided that the following conditions are met:
+
+	1. Redistributions of source code must retain the above copyright notice,
+	   this list of conditions and the following disclaimer.
+
+	2. Redistributions in binary form must reproduce the above copyright
+	   notice, this list of conditions and the following disclaimer in the
+	   documentation and/or other materials provided with the distribution.
+
+	THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+	INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+	AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+	AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+	OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+	SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+	INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+	CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+	ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+	POSSIBILITY OF SUCH DAMAGE.
+*/
+require_once('config.inc');
+require_once('service-utils.inc');
+require_once('util.inc');
+/* include_once('guiconfig.inc'); is needed for clear_log_file() during package installation while booting.
+ * However, guiconfig.inc includes authgui.inc which requires a valid php session_auth() and exits when not found.
+ * So we include the function here.
+*/
+if (!function_exists('clear_log_file')) {
+
+	function clear_log_file($logfile = "/var/log/system.log", $restart_syslogd = true) {
+		require_once('system.inc');
+		global $config;
+		if ($restart_syslogd) {
+			exec("/usr/bin/killall syslogd");
+		}
+		if (isset($config['system']['disablesyslogclog'])) {
+			unlink($logfile);
+			touch($logfile);
+		} else {
+			$log_size = isset($config['syslog']['logfilesize']) ? $config['syslog']['logfilesize'] : "511488";
+			if (isset($config['system']['usefifolog'])) {
+				exec("/usr/sbin/fifolog_create -s {$log_size} " . escapeshellarg($logfile));
+			} else {
+				exec("/usr/local/sbin/clog -i -s {$log_size} " . escapeshellarg($logfile));
+			}
+		}
+		if ($restart_syslogd) {
+			system_syslogd_start();
+		}
+	}
+}
+
+function tinc_save() {
+	global $config, $configpath;
+	$configpath = '/usr/local/etc/tinc';
+
+	conf_mount_rw();
+
+	rename("{$configpath}", "{$configpath}.old");
+	safe_mkdir("{$configpath}");
+	safe_mkdir("{$configpath}/hosts");
+	touch("{$configpath}/WARNING-ENTIRE_DIRECTORY_ERASED_ON_SAVE_FROM_GUI");
```

Comment:
I don't thing this is necessary, users shoudn't touch config files manually on pfSense at all